### PR TITLE
Multipath conf fix

### PR DESCRIPTION
**Pull request by Naresh-ibm**

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
localhost:~/avocado-fvt-wrapper/tests/avocado-misc-tests/io/disk # avocado run multipath_test.py -m multipath_test.py.data/multipath_test.yaml 
JOB ID     : b262275f8619fcfcbbb3051750205885b28a7344
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2021-09-19T22.43-b262275/job.log
 (01/21) multipath_test.py:MultipathTest.test;run-policy-queue-length-a8e5: PASS (201.40 s)
 (02/21) multipath_test.py:MultipathTest.test;run-policy-service-time-7a4f: PASS (201.38 s)
 (03/21) multipath_test.py:MultipathTest.test;run-policy-round-robin-84cb: PASS (201.63 s)
 (04/21) multipath_test.py:MultipathTest.test_fail_reinstate_individual_paths;run-policy-queue-length-a8e5: PASS (43.25 s)
 (05/21) multipath_test.py:MultipathTest.test_fail_reinstate_individual_paths;run-policy-service-time-7a4f: PASS (43.24 s)
 (06/21) multipath_test.py:MultipathTest.test_fail_reinstate_individual_paths;run-policy-round-robin-84cb: PASS (43.26 s)
 (07/21) multipath_test.py:MultipathTest.test_io_run_on_single_path;run-policy-queue-length-a8e5: PASS (394.06 s)
 (08/21) multipath_test.py:MultipathTest.test_io_run_on_single_path;run-policy-service-time-7a4f: PASS (390.58 s)
 (09/21) multipath_test.py:MultipathTest.test_io_run_on_single_path;run-policy-round-robin-84cb: PASS (394.21 s)
 (10/21) multipath_test.py:MultipathTest.test_failing_reinstate_all_paths;run-policy-queue-length-a8e5: PASS (400.44 s)
 (11/21) multipath_test.py:MultipathTest.test_failing_reinstate_all_paths;run-policy-service-time-7a4f: PASS (404.56 s)
 (12/21) multipath_test.py:MultipathTest.test_failing_reinstate_all_paths;run-policy-round-robin-84cb: PASS (404.64 s)
 (13/21) multipath_test.py:MultipathTest.test_removing_all_paths;run-policy-queue-length-a8e5: PASS (368.25 s)
 (14/21) multipath_test.py:MultipathTest.test_removing_all_paths;run-policy-service-time-7a4f: PASS (368.58 s)
 (15/21) multipath_test.py:MultipathTest.test_removing_all_paths;run-policy-round-robin-84cb: PASS (368.20 s)
 (16/21) multipath_test.py:MultipathTest.test_suspend_resume_individual_mpath;run-policy-queue-length-a8e5: PASS (124.51 s)
 (17/21) multipath_test.py:MultipathTest.test_suspend_resume_individual_mpath;run-policy-service-time-7a4f: PASS (124.66 s)
 (18/21) multipath_test.py:MultipathTest.test_suspend_resume_individual_mpath;run-policy-round-robin-84cb: PASS (124.69 s)
 (19/21) multipath_test.py:MultipathTest.test_suspend_resume_all_mpath;run-policy-queue-length-a8e5: PASS (184.72 s)
 (20/21) multipath_test.py:MultipathTest.test_suspend_resume_all_mpath;run-policy-service-time-7a4f: PASS (184.55 s)
 (21/21) multipath_test.py:MultipathTest.test_suspend_resume_all_mpath;run-policy-round-robin-84cb: PASS (184.92 s)
RESULTS    : PASS 21 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2021-09-19T22.43-b262275/results.html
JOB TIME   : 5168.41 s
localhost:~/avocado-fvt-wrapper/tests/avocado-misc-tests/io/disk #

**Comment by Naresh-ibm** [comment]
[multipath_conf_timing_issue_fix_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/7193748/multipath_conf_timing_issue_fix_job.log)

File: io/disk/multipath_test.py [hunk 1]
```diff
@@ -117,40 +118,97 @@ def test(self):
...
             # Path Selector policy
             self.log.info("changing Selector policy")
             for policy in self.policies:
                 cmd = "path_selector \"%s 0\"" % policy
                 multipath.form_conf_mpath_file(defaults_extra=cmd)
+                time.sleep(5)
                 if multipath.get_policy(path_dic["wwid"]) != policy:
                     msg += "%s for %s fails\n" % (policy, path_dic["wwid"])
 
+            def is_mpath_exists():
```
Comment (narasimhan-v) [inline io/disk/multipath_test.py:132]: This can be renamed to `does_mpath_exist` or `is_mpath_available`

File: io/disk/multipath_test.py [hunk 2]
```diff
@@ -117,40 +118,97 @@ def test(self):
...
+            operation = 'recover'
+            self.log.info("Recovering WWID after Black list.....")
+            multipath.form_conf_mpath_file(defaults_extra=plcy)
+            if wait.wait_for(is_mpath_exists, timeout=10):
+                self.log.info("recovery of %s success" % path_dic["wwid"])
             else:
-                multipath.form_conf_mpath_file(defaults_extra=plcy)
-                if not multipath.device_exists(path_dic["wwid"]):
-                    msg += "Recovery of %s fails\n" % path_dic["wwid"]
+                msg += "Recovery of %s fails\n" % path_dic["wwid"]
```
Comment (narasimhan-v) [inline io/disk/multipath_test.py:178]: Recovery of %s fails after blacklist wwid

File: io/disk/multipath_test.py [hunk 3]
```diff
@@ -117,40 +118,97 @@ def test(self):
...
+                if wait.wait_for(is_path_exists, timeout=10):
+                    self.log.info("block list of %s success" % disk)
+                else:
+                    msg += "block list of %s fails\n" % disk
+                operation = 'recover'
+                multipath.form_conf_mpath_file(defaults_extra=plcy)
+                if wait.wait_for(is_mpath_exists, timeout=10):
+                    self.log.info("recovery of %s success" % disk)
+                else:
+                    msg += "Recovery of %s fails\n" % path_dic["wwid"]
```
Comment (narasimhan-v) [inline io/disk/multipath_test.py:210]: Recovery of %s fails after blacklist sdX

**Review by narasimhan-v — CHANGES_REQUESTED** [review]

**Comment by Naresh-ibm** [comment]
localhost:~/avocado-fvt-wrapper/tests/avocado-misc-tests/io/disk # avocado run multipath_test.py -m multipath_test.py.data/multipath_test.yaml 
JOB ID     : 0cc8fdb0f6475872fef87b58f7575d938d236133
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2021-09-20T13.06-0cc8fdb/job.log
 (01/21) multipath_test.py:MultipathTest.test;run-policy-queue-length-a8e5: PASS (210.24 s)
 (02/21) multipath_test.py:MultipathTest.test;run-policy-service-time-7a4f: PASS (210.24 s)
 (03/21) multipath_test.py:MultipathTest.test;run-policy-round-robin-84cb: PASS (210.19 s)
 (04/21) multipath_test.py:MultipathTest.test_fail_reinstate_individual_paths;run-policy-queue-length-a8e5: PASS (37.99 s)
 (05/21) multipath_test.py:MultipathTest.test_fail_reinstate_individual_paths;run-policy-service-time-7a4f: PASS (43.29 s)
 (06/21) multipath_test.py:MultipathTest.test_fail_reinstate_individual_paths;run-policy-round-robin-84cb: PASS (43.16 s)
 (07/21) multipath_test.py:MultipathTest.test_io_run_on_single_path;run-policy-queue-length-a8e5: PASS (394.18 s)
 (08/21) multipath_test.py:MultipathTest.test_io_run_on_single_path;run-policy-service-time-7a4f: PASS (394.45 s)
 (09/21) multipath_test.py:MultipathTest.test_io_run_on_single_path;run-policy-round-robin-84cb: PASS (394.63 s)
 (10/21) multipath_test.py:MultipathTest.test_failing_reinstate_all_paths;run-policy-queue-length-a8e5: PASS (404.44 s)
 (11/21) multipath_test.py:MultipathTest.test_failing_reinstate_all_paths;run-policy-service-time-7a4f: PASS (404.89 s)
 (12/21) multipath_test.py:MultipathTest.test_failing_reinstate_all_paths;run-policy-round-robin-84cb: PASS (405.21 s)
 (13/21) multipath_test.py:MultipathTest.test_removing_all_paths;run-policy-queue-length-a8e5: PASS (369.12 s)
 (14/21) multipath_test.py:MultipathTest.test_removing_all_paths;run-policy-service-time-7a4f: PASS (368.84 s)
 (15/21) multipath_test.py:MultipathTest.test_removing_all_paths;run-policy-round-robin-84cb: PASS (368.92 s)
 (16/21) multipath_test.py:MultipathTest.test_suspend_resume_individual_mpath;run-policy-queue-length-a8e5: PASS (124.43 s)
 (17/21) multipath_test.py:MultipathTest.test_suspend_resume_individual_mpath;run-policy-service-time-7a4f: PASS (124.26 s)
 (18/21) multipath_test.py:MultipathTest.test_suspend_resume_individual_mpath;run-policy-round-robin-84cb: PASS (124.19 s)
 (19/21) multipath_test.py:MultipathTest.test_suspend_resume_all_mpath;run-policy-queue-length-a8e5: PASS (184.27 s)
 (20/21) multipath_test.py:MultipathTest.test_suspend_resume_all_mpath;run-policy-service-time-7a4f: PASS (184.46 s)
 (21/21) multipath_test.py:MultipathTest.test_suspend_resume_all_mpath;run-policy-round-robin-84cb: PASS (184.65 s)
RESULTS    : PASS 21 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2021-09-20T13.06-0cc8fdb/results.html
JOB TIME   : 5198.84 s
localhost:~/avocado-fvt-wrapper/tests/avocado-misc-tests/io/disk #

**Comment by Naresh-ibm** [comment]
@narasimhan-v changed the cosmetic things. please check and merge.

**Review by narasimhan-v — APPROVED** [review]

**Pull request merged**